osu.Game/Models/RealmBeatmapSet.cs

```diff
@@ -53,7 +53,16 @@ public class RealmBeatmapSet : RealmObject, IHasGuidPrimaryKey, IHasRealmFiles,
         /// <param name="filename">The name of the file to get the storage path of.</param>
         public string? GetPathForFile(string filename) => Files.SingleOrDefault(f => string.Equals(f.Filename, filename, StringComparison.OrdinalIgnoreCase))?.File.StoragePath;
 
-        public override string ToString() => Metadata?.ToString() ?? base.ToString();
+        public bool Equals(IBeatmapSetInfo? other)
+        {
+            if (other is RealmBeatmapSet b)
+                return b.ID == ID;
```

Comment:
Shouldn't this use `.Equals(b)`? Has the `IsManaged` check which looks suspiciously required?